project/apps/web/app/questions/page.tsx

```diff
@@ -0,0 +1,134 @@
+"use client";
+
+import { Suspense, useState } from "react";
+import { Plus } from "lucide-react";
+import { QuestionDto, CreateQuestionDto } from "@repo/dtos/questions";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { QUERY_KEYS } from "@/constants/queryKeys";
+import {
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from "@tanstack/react-query";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import CreateModal from "./components/CreateModal";
+import { toast } from "@/hooks/use-toast";
+import { createQuestion, fetchQuestions } from "@/lib/api/question";
+import Link from "next/link";
+import DifficultyBadge from "@/components/DifficultyBadge";
+import QuestionsSkeleton from "./components/QuestionsSkeleton";
+import EmptyPlaceholder from "./components/EmptyPlaceholder";
+
+const QuestionRepositoryContent = () => {
+  const queryClient = useQueryClient();
+  const [isCreateModalOpen, setCreateModalOpen] = useState(false);
+  const [confirmLoading, setConfirmLoading] = useState(false);
+
+  const { data } = useSuspenseQuery<QuestionDto[]>({
+    queryKey: [QUERY_KEYS.Question],
+    queryFn: fetchQuestions,
+  });
+  const createMutation = useMutation({
+    mutationFn: (newQuestion: CreateQuestionDto) => createQuestion(newQuestion),
+    onMutate: () => setConfirmLoading(true),
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({ queryKey: [QUERY_KEYS.Question] });
+      setCreateModalOpen(false);
+      toast({ variant: "success", title: "Success", description: "Question created successfully"});
+    },
+    onSettled: () => setConfirmLoading(false),
+    onError: (error) => {
+      console.error("Error creating question:", error);
+      toast({ variant: "destructive", title: "Error", description: "Error creating question: " + error });
+    },
```

Comment:
```suggestion
        onError: (error) => {
      const errorMsg = `Error creating question: ${error.message}`;
      console.error(errorMsg);
      toast({
        variant: "destructive",
        title: "Error",
        description: errorMsg,
      });
    },
```

Noted that question-service throws 500 Internal server error now, will create a hotfix PR for this issue

Reply:
Fixed in #15